kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java

```diff
@@ -614,7 +614,9 @@ public void process(BeanDefinition<?> beanDefinition, ExecutableMethod<?, ?> met
                         // ignore for shutdown
                     } finally {
                         try {
-                            if (offsetStrategy != OffsetStrategy.DISABLED) {
+                            if (offsetStrategy != OffsetStrategy.DISABLED &&
+                                   ! ((offsetStrategy == OffsetStrategy.SYNC || offsetStrategy == OffsetStrategy.ASYNC) && failed)) {
```

Comment:
Shouldn't this be `!failed` ?

Reply:
Well its a bit more complex. Please note that the whole expression is negated. Basically it means if offsetStrategy == SYNC or ASYNC AND it is failed than do NOT commit.

   ! ((offsetStrategy == OffsetStrategy.SYNC || offsetStrategy == OffsetStrategy.ASYNC) && failed))

I didn't want to change the existing behavior too much. Only in case of the SYNC or ASYNC it violates the specification. However I believe there should be a discussion about the behavior for the other sync strategies.